segtypes/segment.py

```diff
@@ -325,6 +333,14 @@ def dir(self) -> Path:
         else:
             return self.given_dir
 
+    @property
+    def show_file_boundaries(self) -> bool:
+        if self.parent:
+            # If either the parent or the segment itself needs to show boundaries.
+            return self.find_file_boundaries and self.parent.find_file_boundaries
```

Comment:
```suggestion
            return self.find_file_boundaries or self.parent.find_file_boundaries
```
Given your comment, I think you want this to be a logical "or", but I'm not sure if that's your intent

Reply:
Yeah, the comment is outdated and should have been changed before commit. What happens with `or` is that if the parent has `find_file_boundaries = True` and the segment itself has it to `False`, it still shows for the "child" segment. I changed the logic to `and`, and it gave me the correct outputs when testing. I'll revisit it though, cause now I'm contemplating if I did it right...